packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -11,9 +11,83 @@ import 'framework.dart';
 /// Used by [NotificationListener.onNotification].
+
 typedef NotificationListenerCallback<T extends Notification> = bool Function(T notification);
 
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This example shows a [NotificationListener ] widget
+/// which its type is [ScrollNotification ], when there is scroll
+/// event occurs in the [NestedScrollView],
+/// We want to be notified the current scroll position whether
+/// it's [ScrollStartNotification]or[ScrollEndNotification].
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+/// const List<String> _tabs = ["Months", "Days"];
+/// const List<String> _months = [ "January","February","March", ];
+/// const List<String> _days = [ "Sunday", "Monday","Tuesday", ];
+///   return DefaultTabController(
+///     length: _tabs.length,
+///     child: Scaffold(
+///       //Listens the scroll events and returns the current position.
+///       body: NotificationListener<ScrollNotification>(
+///         onNotification: (scrollNotification) {
+///           if (scrollNotification is ScrollStartNotification) {
+///             print('Scrolling has Started');
+///           } else if (scrollNotification is ScrollEndNotification) {
+///             print("Scrolling has ended");
+///           }
+///          // Return true to cancel the notification bubbling.
+///           return true;
+///         },
+///         child: NestedScrollView(
+///           headerSliverBuilder:
+///               (BuildContext context, bool innerBoxIsScrolled) {
+///             return <Widget>[
+///               SliverAppBar(
+///                 title: const Text("World"),
+///                 pinned: true,
+///                 floating: true,
+///                 bottom: TabBar(
+///                   tabs: _tabs.map((name) => Tab(text: name)).toList(),
+///                 ),
+///               ),
+///             ];
+///           },
+///           body: TabBarView(
+///             children: [
+///               ListView.builder(
+///                 itemCount: _months.length,
+///                 itemBuilder: (context, index) {
+///                   return ListTile(title: Text(_months[index]));
+///                 },
+///               ),
+///               ListView.builder(
+///                 itemCount: _days.length,
+///                 itemBuilder: (context, index) {
+///                   return ListTile(title: Text(_days[index]));
+///                 },
+///               ),
+///            ],
+///           ),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+/// ### [ScrollNotification ] which describes the notification lifecycle.
+/// ### [ScrollStartNotification] which returns the start position of scrolling.
+/// ### [ScrollEndNotification] which returns the end position of scrolling.
+/// ### [NestedScrollView] which creates a nested scroll view.
```

Comment:
The formatting here isn't quite right. See other docs for how this is formatted.